homeassistant/components/ios/__init__.py

```diff
@@ -247,14 +249,17 @@ def device_name_for_push_id(hass, push_id):
     return None
 
 
-async def async_setup(hass, config):
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the iOS component."""
     conf = config.get(DOMAIN)
 
     ios_config = await hass.async_add_executor_job(
         load_json, hass.config.path(CONFIGURATION_FILE)
     )
 
+    if TYPE_CHECKING:
+        assert isinstance(ios_config, dict)
```

Comment:
```suggestion
    ios_config: dict = await hass.async_add_executor_job(
        load_json, hass.config.path(CONFIGURATION_FILE)
    )
```

Reply:
That is what I originally used, but `load_json` returns `list | dict` and it triggers this error:
```
homeassistant/components/ios/__init__.py:257: error: Argument 1 to "async_add_executor_job" 
of "HomeAssistant" has incompatible type 
"Callable[[str, Union[List[Any], Dict[Any, Any], None]], Union[List[Any], Dict[Any, Any]]]"; 
expected "Callable[..., Dict[Any, Any]]"  [arg-type]
Found 1 error in 1 file (checked 5 source files)
```

Reply:
You are right 👍